pkg/synthetic/ticket.go

```diff
@@ -0,0 +1,186 @@
+		structs.Number(4),
+		structs.Number(5),
+	}, []int64{
+		50, 30, 13, 5, 2,
+	}),
+	BadArtificalLatency(
+		"region.europe-east1",
+		"region.europe-west1",
+		"region.europe-west2",
+		"region.europe-west3",
+		"region.europe-west4",
+	),
+}
+
+// GenFunc defines a function which generates values for ticket properties.
+type GenFunc func(func(string, *structpb.Value))
+
+// CustomTicket creates a ticket using a custom list of generators.  If multiple
+// generators try to add a value to the same property name, CustomTicket will
+// panic.
+func CustomTicket(generators ...GenFunc) *pb.Ticket {
+	properties := structs.Struct{}
+	add := func(field string, value *structpb.Value) {
+		if _, ok := properties[field]; ok {
+			panic(fmt.Sprintf("Field %s populated with two differnet values.", field))
+		}
+		properties[field] = value
+	}
+
+	for _, gen := range generators {
+		gen(add)
+	}
+
+	return &pb.Ticket{Properties: properties.S()}
+}
+
+// Normal creates a generator function which sets a field to a value following
+// a normal distribution.  The value is clamped by min and max.
+func Normal(field string, avg, stdev, min, max float64) GenFunc {
+	return func(add func(string, *structpb.Value)) {
+		v := rand.NormFloat64()*stdev + avg
+		if v > max {
+			v = max
+		}
+		if v < min {
+			v = min
+		}
+
+		add(field, structs.Number(v))
+	}
+}
+
+// FieldChoice creates a generator function which uses a defined weights to
+// choose a random property field to set, and sets that property to 1.
+func FieldChoice(weights map[string]int64) GenFunc {
+	options := make([]string, 0, len(weights))
+	relativeWeights := make([]int64, 0, len(weights))
+	for k, v := range weights {
+		options = append(options, k)
+		relativeWeights = append(relativeWeights, v)
+	}
+
+	choice := weightedChoice(relativeWeights)
+	return func(add func(string, *structpb.Value)) {
+		add(options[choice()], structs.Number(1))
+	}
+}
+
+// ValueChoice creates a generator function which sets the field to a random
+// value chosen by the weights.
+func ValueChoice(field string, values []*structpb.Value, weights []int64) GenFunc {
+	if len(values) != len(weights) {
+		panic("Values and weights must have same length")
+	}
+	choice := weightedChoice(weights)
+	return func(add func(string, *structpb.Value)) {
+		add(field, values[choice()])
+	}
+}
+
+func weightedChoice(weights []int64) func() int {
+	{
+		cp := make([]int64, len(weights))
+		copy(cp, weights)
+		weights = cp
+	}
+	totalWeight := int64(0)
+	for i, v := range weights {
+		weights[i] = totalWeight
+		totalWeight += v
+	}
+	return func() int {
+		r := rand.Int63n(totalWeight)
+		low := 0
+		high := len(weights) - 1
+		for low < high {
+			mid := low + ((high - low) / 2)
+			if r >= weights[mid] {
+				if r < weights[mid+1] {
+					return mid
+				}
+				low = mid + 1
+			} else {
+				high = mid - 1
+			}
+		}
+		return low
+	}
+}
+
+// BadArtificalLatency creates some latency metrics from a list of regions.
```

Comment:
Since this is fake anyways - why BadArtificalLatency? Just Latency() to map with other helpers? Or furthermore - just a randomly ordered list of regions? reusing one of the helpers you already have? If its bad - might as well have it simple and bad?

Reply:
Using any latency number raw is probably bad... This at least states that on the tin.  I don't want people to use this expecting that this will give reasonable latency numbers.

Reply:
Going with some simple approach - like simply ordering regions latency wise (no actual latency in ms - just an order) - will be ok - IMO - that way, you dont have latency numbers but take into account the fact that a player will have some order of preference for the backends latency wise. 

That would be an approximation. I think that expectation from a 'fake' player is good enough. Or - if we think its just bad - then we can simply drop it. If we think this property is 'bad' even for a 'fake' player, might as well drop it.